### PR TITLE
Add TLS R/W mutex to rfbClient

### PR DESCRIPTION
**Pull request by gujjwal00**
I have added a comment to indicate its internal use but to allow this type of modifications in future we could create a separate 'internal' structure.

### CODE REVIEW

File: rfb/rfbclient.h [hunk 1]
```diff
@@ -378,6 +379,12 @@ typedef struct _rfbClient {
 	/** The TLS session for Anonymous TLS and VeNCrypt */
 	void* tlsSession;
 
+	/**
+	 * Mutex to protect concurrent TLS read/write.
+	 * For internal use only.
+	 */
+	MUTEX(tlsRwMutex);
```
Comment (bk138) [inline rfb/rfbclient.h:386]: please make sure this goes to the end of the struct to keep ABI compatibility

Comment (gujjwal00) [reply rfb/rfbclient.h]: Done!

**Comment by bk138** [comment]
Thanks! It doesn't do very much ATM though ;-)

**Comment by gujjwal00** [comment]
Yes, I am planning separate PRs for OpenSSL & GnuTLS.

I was going to create a single PR but then saw the new branch and decided to create smaller ones.

**Comment by bk138** [comment]
Let's make bigger PRs that contain sets of functionaliy :-) We can use this one via force-push.

**Comment by gujjwal00** [comment]
@bk138 Do we have any minimum version requirement for OpenSSL?

**Comment by bk138** [comment]
> @bk138 Do we have any minimum version requirement for OpenSSL?

The baseline is 1.0.2g, me thinks.

**Comment by gujjwal00** [comment]
Would it be possible to require 1.1.0 at-least (if apps want thread safety)? 

I don't want to break existing apps (so don't shoot me for this 😄). I am just asking for you opinion.

**Comment by bk138** [comment]
> Would it be possible to require 1.1.0 at-least (if apps want thread safety)?
> 
> I don't want to break existing apps (so don't shoot me for this smile). I am just asking for your opinion.

I would like to not break stuff for exisitng users as of now, so rather keep compat with 1.0.x

**Comment by gujjwal00** [comment]
I will cleanup existing mutex defines in `tls_openssl.c` in next PR.

**Pull request merged**